code/smart-mobility/smart-mobility.js

```diff
@@ -177,3 +177,129 @@ $(document).on("knack-scene-render.scene_281", function () {
 $(document).on("knack-scene-render.scene_284", function () {
   disableBreadCrumbsNonAdmin();
 });
+
+  //***MANAGE WALLET AND MAKE PAYMENT BUTTONS***
+$(document).on('knack-view-render.view_804', function(event, page) {
+  // create large button on go back to Portal page
+  bigButton('manage-wallet', 'view_804', "https://austin-tx-austin-tx.uat.cityba.se/welcome", "credit-card", "Manage Wallet");
+  // bigButton('manage-wallet', 'view_804', "https://atd.knack.com/smart-mobility#living-streets-applicant-portal/", "credit-card", "Manage Wallet");
+});
+$(document).on('knack-view-render.view_806', function(event, page) {
+  // create large button on go back to Portal page
+  // *** see the getCitybaseButton function for the invocation of the make payment bigButton function
+  //   bigButton('make-payment', 'view_806', "https://atd.knack.com/smart-mobility#living-streets-applicant-portal/", "arrow-right", "Make Payment");
+});
+
+/** 
+ * Citybase Integration
+ */
+
+function getCitybaseButton(payload, viewId) {
+  fetch(
+    "https://invoice-service-austin-tx.uat.cityba.se/invoices/austin_tx_transportation/street_banner",
+    {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(payload),
+    },
+  )
+    .then((response) => {
+      console.log(response.url);
+      bigButton(
+        "make-payment",
+        viewId,
+        response.url,
+        "arrow-right",
+        "Make Payment",
+      );
+    })
+    .catch((error) => {
+      console.error("Error:", error);
+    });
+}
+
+var knackUserToken = Knack.getUserToken();
+
+var headers = {
+  "X-Knack-Application-Id": "618ad3322d11b4002169a6f9",
+  "X-Knack-REST-API-KEY": "knack",
+  Authorization: knackUserToken,
+  "content-type": "application/json",
+};
+
+// After the transactions table loads
+$(document).on("knack-view-render.view_817", function (event, page, view) {
+  $('#view_817').hide()
+  var transactionRecord = view[0];
+
+  var payload = {
+    allowed_payment_methods: ["CARD", "BANK"],
+    cancel_url: {
+      url: window.location.href,
+      label: "Cancel",
+    },
+    return_url: {
+      url: window.location.href,
+      label: "Continue",
+    },
+  };
+
+  // make sure record exists
+  if (transactionRecord) {
+    payload["line_items"] = [
+      {
+        description: transactionRecord["field_819"],
+        amount: transactionRecord["field_833_raw"] * 100,
+        sub_description: transactionRecord["field_849"],
+        custom_attributes: [
+          {
+            key: "knack_record_id",
+            value: transactionRecord["id"],
+          },
+          {
+            key: "invoice_number",
+            value: transactionRecord["field_814"],
+          },
```

Comment:
These IDs are always `Number` I figure based on the lack of quotes, but maybe keep that pattern from below where you are using `String()` going by using `Number()` around these values -- not a big deal if you have high confidence in the incoming data.